libs/vkd3d-common/profiling.c

```diff
@@ -192,4 +226,34 @@ void vkd3d_profiling_notify_work(unsigned int index,
     spinlock_release(lock);
 }
 
+#ifdef TRACY_ENABLE
+
+void ___vkd3d_set_thread_name( const char* name )
+{
+    if(pfn_tracy_set_thread_name)
+        pfn_tracy_set_thread_name(name);
+}
+
+void ___vkd3d_emit_frame_mark()
+{
+    if(pfn_tracy_emit_frame_mark)
+        pfn_tracy_emit_frame_mark(0);
+}
+
+TracyCZoneCtx ___vkd3d_emit_zone_begin( const struct ___tracy_source_location_data* srcloc, int active )
+{
+    static TracyCZoneCtx ctx;
```

Comment:
What if this is used from multiple threads concurrently?

Reply:
Tracys functions are all thread safe. The default return value shouldn't be modified by the calling code. If the tracy dll is not found, ctx would not be passed to ___tracy_emit_zone_end either.